pycodestyle.py

```diff
@@ -1995,8 +1995,9 @@ def __init__(self, *args, **kwargs):
         # build options from dict
         options_dict = dict(*args, **kwargs)
         arglist = None if parse_argv else options_dict.get('paths', None)
+        verbose = options_dict.get('verbose', False)
```

Comment:
So we allow `verbose` to be in the options dictionary, yes? If it is, it was passed here explicitly. Otherwise, (in the case where someone isn't directly using the StyleGuide but is instead using the CLI) we'll parse it from the arguments. Below here, we do `options.__dict__.update(options_dict)` which means that after we've processed options, we update the values unconditionally so long as `options_dict` isn't an empty dictionary. That's all you should need.

Reply:
I'm a bit confused, what is the change you are suggesting I make?

Reply:
So, in your use case, you are going to do:

```
StyleGuide(verbose=True)
```

Which means that `options_dict` will look at least like `{'verbose': True}`. After we call `process_options` on L1999,L2000 we say `if options_dict: options.__dict__.update(options_dict)` which means that regardless of what's passed on the CLI we will set `verbose` to True. That means that you don't need to modify process_options.

I suspect the problem you're actually encountering and trying to solve by this PR is that everything in pycodestyle expects `verbose` to be an integer. See, as one example, this line: https://github.com/PyCQA/pycodestyle/blob/19e85d98eca9dd41d949b60a47c0804e672327a2/pycodestyle.py#L555

If you're not seeing the output you expect when using the Python API, it's because `True` is being treated as `1` by Python and you're only getting basic verbosity levels. 

```
~ ❯❯❯ python -c 'print(True + 1)'
2
~ ❯❯❯ python3 -c 'print(True + 1)'
2
~ ❯❯❯ python -c 'print(True >= 1)'
True
~ ❯❯❯ python -c 'print(True >= 2)'
False
```

Reply:
I did not know that the `verbose` is meant to be a number, so I've changed that accordingly.  

I do need to modify `process_options` somehow due to the fact that `process_options` calls `read_config`
https://github.com/PyCQA/pycodestyle/blob/19e85d98eca9dd41d949b60a47c0804e672327a2/pycodestyle.py#L2343
And `read_config` checks for `options.verbose`
https://github.com/PyCQA/pycodestyle/blob/19e85d98eca9dd41d949b60a47c0804e672327a2/pycodestyle.py#L2251
Which does not get set until L1999 & L2000, after `read_config` is called, unless you instantiate `StyleGuide(paths=['-v'])`, which does set `options.verbose` to `1`. 

I do not think it makes sense to have to pass a CLI argument to the styleguide object to achieve full verbosity, as that is what the verbose argument is for.

Reply:
Sorry, I think you had mentioned this before. Thank you for reminding me.

So then the changes necessary here are the following:

```py
verbose = options_dict.pop('verbose', 0)
```

Then below pass it to `process_options` as you are. You can make the change there of having `verbose=None` in the argument list and only if `verbose is not None` will you set it on the `options` object.

I think a better path forward, however, is to refactor pycodestyle and remove config reading from CLI argument processing but that's not for this PR.